packages/core/src/browser/extension-open-handler.ts

```diff
@@ -0,0 +1,63 @@
+// *****************************************************************************
+// Copyright (C) 2024 STMicroelectronics.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { injectable } from 'inversify';
+import { MaybePromise, URI } from '../common';
+import { OpenHandler, OpenerOptions } from './opener-service';
+
+export interface UriHandler {
+    canHandleURI(uri: URI): boolean;
+    handleUri(uri: URI): Promise<boolean>;
+}
+
+@injectable()
+export class ExtensionOpenHandler implements OpenHandler {
+
+    readonly id = 'extensionsURIHandlers';
+
+    private providers = new Map<string, UriHandler>();
```

Comment:
If it stores handlers, it should be called `handlers` ideally `handlersByAuthority`.